src/main/scala/com/databricks/spark/avro/AvroRelation.scala

```diff
@@ -106,15 +135,28 @@ case class AvroRelation(location: String)(@transient val sqlContext: SQLContext)
           ArrayType(schemaType.dataType, containsNull = schemaType.nullable),
           nullable = false)
 
-      case UNION => avroSchema.getTypes.toSeq match {
-        case Seq(t1, t2) if t1.getType == NULL => toSqlType(t2).copy(nullable = true)
-        case Seq(t1, t2) if t2.getType == NULL => toSqlType(t1).copy(nullable = true)
-        case other =>
-          sys.error(s"Union types with anything other than null not supported: $other")
+      case MAP =>
+        val schemaType = toSqlType(avroSchema.getValueType)
+        SchemaType(
+          MapType(StringType, schemaType.dataType, valueContainsNull = schemaType.nullable),
+          nullable = false)
+
+      case UNION => {
+        if (avroSchema.getTypes.map(x => x.getType).contains(NULL)) {
```

Comment:
nit: should use same variable name here and in the filter (or just `_`). Also, even more nitty, but for consistency I might also use `.exists(_.getType == NULL)` so the logic looks more analogous.